core/src/main/java/com/lafaspot/imapnio/client/ImapClientRespReader.java

```diff
@@ -0,0 +1,93 @@
+package com.lafaspot.imapnio.client;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.DelimiterBasedFrameDecoder;
+import io.netty.handler.codec.Delimiters;
+
+import com.sun.mail.util.ASCIIUtility;
+
+/**
+ * Basic response reader, read response from channel and decode based on line delimiter, also could handle imap literal response.
+ *
+ * @author kaituo
+ *
+ */
+public class ImapClientRespReader extends DelimiterBasedFrameDecoder {
+
+    /** Literal response remaining bytes size. */
+    private int literalCount;
+    /** Literal response buffer. */
+    private ByteBuf literalBuffer;
+
+    /** Minimal literal response  length. */
+    private static final int MINIMAL_LITERAL_LENGTH = 5;
+    /** Literal response '}' index from tail. */
+    private static final int LITERAL_INDEX_FROM_TAIL = 3;
+
+    /**
+     * Constructor of imap client response reader.
+     *
+     * @param maxLineLength maximum response line length
+     */
+    public ImapClientRespReader(final int maxLineLength) {
+        super(maxLineLength, false, Delimiters.lineDelimiter());
```

Comment:
Line delimiter allows newline or (carriage return-newline). Imap protocol is always delimited by carriage return - newline. Change to use only carriage return - newline.